### ISSUE TITLE
Question about Temperature for disp loss

### ISSUE THREAD

**Issue by jihopark00** [raywang4/DispLoss#7]
Hi thanks for your great work!

I noticed that in your implementation of the dispersive loss (InfoNCE-L2 variant), the distance is scaled by the feature dimension (z.shape[1])
My understanding is that this scaling helps normalize the L2 distance because it increases with dimensionality. 
However, I have not been able to find a reference paper that explicitly uses this dimension-dependent temperature in Contrastive Learning.

Would you be able to provide a reference work where this temperature approach is discussed? 

```
def disp_loss(self, z): # Dispersive Loss implementation (InfoNCE-L2 variant)
  z = z.reshape((z.shape[0],-1)) # flatten
  diff = th.nn.functional.pdist(z).pow(2)/z.shape[1] # pairwise distance
  diff = th.concat((diff, diff, th.zeros(z.shape[0]).cuda()))  # match JAX implementation of full BxB matrix
  return th.log(th.exp(-diff).mean()) # calculate loss
```

Thank you!

**Comment by raywang4** [comment]
Hi, and thanks for the interest in our work. This scaling factor is an engineering design and can be conceptually merged into the temperature as seen in typical InfoNCE loss. To my limited knowledge, I am not aware of other works that exactly does this. We'd also be more than happy to learn about related works if there's any.

**Comment by jihopark00** [comment]
Oh I see. I was just looking for related works that I could study, but I couldn't find any.
Thank you for your kind and prompt response!
